src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,592 @@
+    learningPeriod(learningPeriod),
+    reestimationPeriod(reestimationPeriod),
+    averagedAnomaly(aggregationWindow) {
+        iteration = 0;
+        probationaryPeriod = learningPeriod+estimationSamples;
+        assert(historicWindowSize >= estimationSamples); // cerr << "estimationSamples exceeds historicWindowSize";
+        assert(aggregationWindow < reestimationPeriod && reestimationPeriod < historicWindowSize);
+        
+        runningAverageAnomalies.set_capacity(historicWindowSize); 
+        runningLikelihoods.set_capacity(historicWindowSize);
+        runningRawAnomalyScores.set_capacity(historicWindowSize);
+        runningRawValues.set_capacity(historicWindowSize);
+        assert(runningLikelihoods.capacity() == historicWindowSize);
+    }
+
+    
+    Real anomalyProbability(Real rawValue, Real anomalyScore, int timestamp=-1) { //TODO "rawValue" could be typed to T rawValue; //TODO is rawValue ever used?
+    /**
+    Compute the probability that the current value plus anomaly score represents
+    an anomaly given the historical distribution of anomaly scores. The closer
+    the number is to 1, the higher the chance it is an anomaly.
+
+    @param value - the current metric ("raw") input value, eg. "orange", or
+                   '21.2' (deg. Celsius), ...
+    @param anomalyScore - the current anomaly score
+    @param timestamp - (optional) timestamp of the ocurrence,
+                       default (-1) results in using iteration step.
+    @return the anomalyLikelihood for this record.
+    **/
+    Real likelihood = 0.5f;
+
+    if (timestamp == -1) {
+      timestamp = this->iteration;
+    }
+
+    // store into relevant variables
+    this->runningRawAnomalyScores.push_back(anomalyScore); //TODO implement these vectors as sliding windows
+    auto newAvg = this->averagedAnomaly.compute(anomalyScore); 
+    this->runningAverageAnomalies.push_back(newAvg);
+    this->iteration++;
+    this->runningRawValues.push_back(rawValue); 
+    this->runningLikelihoods.push_back(likelihood);
+    
+    // We ignore the first probationaryPeriod data points - as we cannot reliably compute distribution statistics for estimating likelihood
+    if (this->iteration < this->probationaryPeriod) {
+      return 0.5f;
+    } //else {
+
+    auto anomalies = circularBufferToVector(this->runningAverageAnomalies); 
+    
+      // On a rolling basis we re-estimate the distribution
+      if ( this->iteration == 0 || (this->iteration % this->reestimationPeriod) == 0  || this->distribution.name == "unknown" ) {
+
+        auto numSkipRecords = this->calcSkipRecords_(this->iteration, this->runningAverageAnomalies.capacity(), this->learningPeriod); //FIXME this erase (numSkipRecords) is a problem when we use sliding window (as opposed to vector)! - should we skip only once on beginning, or on each call of this fn?
+        estimateAnomalyLikelihoods(anomalies, numSkipRecords);  // called to update this->distribution;  
```

Comment:
FIXME this erase (numSkipRecords) - should we skip only once on beginning, or on each call of this fn?